apps/mochi-web/components/settings/general/TransactionLimit.tsx

```diff
@@ -0,0 +1,211 @@
+      type: 'evm',
+    },
+  },
+]
+
+interface Props {
+  control: Control<GeneralFormValue>
+  watch: UseFormWatch<GeneralFormValue>
+}
+
+export const TransactionLimit = ({ control, watch }: Props) => {
+  const { fields, append, remove, update } = useFieldArray({
+    control,
+    name: 'transactionLimit',
+  })
+  const enableTransactionLimit = watch('enableTransactionLimit')
+
+  return (
+    <div className="flex flex-col w-full max-w-md space-y-2">
+      <div className="flex items-center justify-between">
+        <FormLabel>Set the limit for transaction</FormLabel>
+        <Controller
+          name="enableTransactionLimit"
+          control={control}
+          render={({ field: { value, onChange, ...rest } }) => (
+            <Switch
+              {...rest}
+              checked={value}
+              onCheckedChange={(checked) => onChange(checked)}
+            />
+          )}
+        />
+      </div>
+      {fields.map((each, index) => (
+        <Controller
+          key={each.id}
+          name={`transactionLimit.${index}`}
+          control={control}
+          render={({ field }) => {
+            const minToken = tokenList.find(
+              (token) => token.id === field.value.minToken,
+            )
+            const minAmount = mochiUtils.formatDigit({
+              value: field.value.minAmount,
+            })
+            const minUsd =
+              Number(field.value.minAmount) * (minToken?.price || 0)
+            const minUsdValue = mochiUtils.formatUsdDigit(minUsd)
+            const maxToken = tokenList.find(
+              (token) => token.id === field.value.maxToken,
+            )
+            const maxAmount = mochiUtils.formatDigit({
+              value: field.value.maxAmount,
+            })
+            const maxUsd =
+              Number(field.value.maxAmount) * (maxToken?.price || 0)
+            const maxUsdValue = mochiUtils.formatUsdDigit(maxUsd)
+            return (
+              <div
+                className={clsx(
+                  'flex items-center border rounded-md border-divider shadow-input',
+                  { 'opacity-25 pointer-events-none': !enableTransactionLimit },
+                )}
+              >
+                <div className="flex-1 p-2 pl-10 overflow-hidden">
+                  <Typography level="h8">
+                    {
+                      actionList.find(
+                        (action) => action.key === field.value.action,
+                      )?.label
+                    }
+                  </Typography>
+                  <div className="flex flex-wrap">
+                    <Typography level="p5" className="mr-2">
+                      Minimun value: {minAmount} {minToken?.symbol} (&#8776;{' '}
+                      {minUsdValue}),
+                    </Typography>
+                    <Typography level="p5">
+                      Maximun value: {maxAmount} {maxToken?.symbol} (&#8776;{' '}
+                      {maxUsdValue}),
```

Comment:
This comma looks weird. Was this an intention?

<img width="305" alt="image" src="https://github.com/consolelabs/mochi-ui/assets/12707960/c3251ca9-1839-4217-b13d-9fa7c8988e74">